exchange/bitswap/bitswap.go

```diff
@@ -375,6 +373,10 @@ func (bs *Bitswap) ReceiveMessage(ctx context.Context, p peer.ID, incoming bsmsg
 			k := b.Cid()
 			log.Event(ctx, "Bitswap.GetBlockRequest.End", k)
 
+			for _, ses := range bs.SessionsForBlock(k) {
+				ses.receiveBlockFrom(p, b)
```

Comment:
Unless I'm mistaken, this is the *only* place that feeds blocks into the session. If a block gets evicted from the interested set *before* we find it, we'll never forward it to the session.